### PR TITLE
Remove two opportunities for nullptr derefs

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by jimingham — APPROVED** [review]
Not sure why someone would tell us to add an empty module, but it is always good to check...

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp [hunk 1]
```diff
@@ -504,9 +504,11 @@ void SwiftLanguageRuntimeImpl::ProcessModulesToAdd() {
   // Add all defered modules to reflection context that were added to
   // the target since this SwiftLanguageRuntime was created.
   modules_to_add_snapshot.ForEach([&](const ModuleSP &module_sp) -> bool {
-    AddModuleToReflectionContext(module_sp);
-    progress.Increment(++completion,
-                       module_sp->GetFileSpec().GetFilename().AsCString());
+    if (module_sp) {
+      AddModuleToReflectionContext(module_sp);
+      progress.Increment(++completion,
+                         (std::string)module_sp->GetFileSpec().GetFilename());
```
Comment (felipepiovezan) [inline lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp]: I strongly suggest avoiding C-style casts, few people can actually state the rules the compiler follow for those in C++...  we can just do: `module_sp->GetFileSpec().GetFilename().GetString()`

**Comment by medismailben** [comment]
LGTM

**Pull request merged**